### PR TITLE
Update dependencies to use Jakarta EE 9 versions of EL and Servlet

### PR DESCRIPTION
**Pull request by markt-asf**
Mainly the API updates but some other minor fixes (build warning, copyright date for Javadoc, POM clean-up)

### CODE REVIEW

File: api/pom.xml [hunk 1]
```diff
@@ -116,62 +161,78 @@
                     </execution>
                 </executions>
             </plugin>
+
+            <!-- Adds the manifest file created by the org.apache.felix:maven-bundle-plugin -->
             <plugin>
-                <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-jar-plugin</artifactId>
-                <version>3.1.0</version>
+                <version>2.4</version>
```
Comment (arjantijms) [inline api/pom.xml]: Is this one intended to be downgraded?

Comment (markt-asf) [reply api/pom.xml]: No. I was aligning JSP with EL and didn't notice the downgrade. I'll go back to EL, check I can update it there and then remove this change from the PR. Thanks for catching this.

**Comment by markt-asf** [comment]
Not yet. We need the updated JSP API in Maven Central and then we can update the implementation to the new APIs and do the renames in the code. The implementation changes in this PR are essentially clean-up and fixing minor errors.

**Comment by markt-asf** [comment]
Tx

**Pull request merged**